Enabled flag for entire Content pack

f4iTh opened the pull request:
The feature may not be useful for the _average player_, but it might be a life saver for some modders. Being able to disable a content pack without having to remove any of the files can be convenient.

What would be a nice addition to this however, would be for SMAPI to read through the _content.json_ files of each Content pack first so it wouldn't load the ones that will be disabled for nothing. 
![screenshot_531](https://user-images.githubusercontent.com/34132723/36879906-9faf51d4-1dce-11e8-9532-0ecb8f392f0c.png)
A simple photo example. 

This would be a nice addition but I don't think it's exactly _needed_. 
Useful? Yes. Must have? Not really.

f4iTh committed:
Merge pull request #1 from Pathoschild/develop

Keeping this updated

f4iTh committed:
Merge pull request #2 from Pathoschild/develop

Updating fork

f4iTh committed:
Update ContentConfig.cs

f4iTh committed:
Enabled flag support for entire Content pack

Pathoschild commented:
Thanks for the pull request!

I'm not sure we need an `Enabled` flag for the whole content pack though. I want to keep the JSON file as simple as possible, so it's easy for modders to learn. Disabling the entire content pack seems to me like something that should be handled by a mod manager, SMAPI (e.g. an `Enabled` field in `manifest.json`), or the player (zip and delete the folder).

f4iTh committed:
Merge pull request #3 from Pathoschild/develop

Updating fork

f4iTh committed:
fixed NullReferenceException (#201)

Tested in game with and without a tractor: No more NullReferenceException on my side.
